pkg/api/upgradelog/handler.go

```diff
@@ -82,6 +83,33 @@ type Handler struct {
 	upgradeLogClient ctlharvesterv1.UpgradeLogClient
 }
 
+func (h Handler) IsMatchedResource(resource subresource.Resource, method string) bool {
+	if resource.Name != subresource.UpgradeLogs.Resource {
+		return false
+	}
+
+	if resource.SubResource == downloadArchiveLink && method == http.MethodGet {
+		return true
+	}
+
+	if resource.SubResource == generateArchiveAction && method == http.MethodPost {
+		return true
+	}
```

Comment:
Is there ever a use-case where we'd need the generate action but not the download action? If not, the generate action would be superfluous.